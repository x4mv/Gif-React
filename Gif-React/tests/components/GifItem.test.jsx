import { Screen, render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"


describe('Probando <GifItem/>', () => { 

    const title = 'saitama'
    const url = 'https://google.com/'
    test('Is required a title and a url and match snapshot', () => { 
        
        const { container } = render(
            <GifItem
                title={title}
                url={url}
            />
        )
        expect(container).toMatchSnapshot()


    })

    test('debe mostrar la imagen con url y alt', () => { 

        render(<GifItem
            title={title}
            url={url}
        />)

        //* Comprobando que el url sea igual al que le mando
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Probandp que el titulo exista', () => { 

        render(<GifItem
            title={title}
            url={url}
        />)

        expect(title).toBeTruthy()
    })
})