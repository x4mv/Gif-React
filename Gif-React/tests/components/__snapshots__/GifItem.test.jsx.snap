// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando <GifItem/> Is required a title and a url and match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://google.com/"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
